doc/webhook.md

```diff
@@ -0,0 +1,47 @@
+## Admission Webhook
+
+[Admission webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/) are HTTP callbacks that receive admission requests and do something with them.
+There are  two webhooks [ValidatingAdmissionWebhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook) and 
+[MutatingAdmissionWebhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook) which are basically 
+doing the same thing except MutatingAdmissionWebhook can modify the requests. In our case, we use MutatingAdmissionWebhook.
+
+In the Pravega operator repo, we are leveraging the webhook implementation from controller-runtime package, here is the [GoDoc](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/webhook). 
+In detail, there are two steps that developers need to do 1) create webhook server and 2) implement the handler.
+The webhook server registers webhook configuration with the apiserver and creates an HTTP server to route requests to the handlers.
+The server is behind a Kubernetes Service and provides a certificate to the apiserver when serving requests. The kubebuilder has a detailed instruction of 
+building a webhook, see [here](https://github.com/kubernetes-sigs/kubebuilder/blob/86026527c754a144defa6474af6fb352143b9270/docs/book/beyond_basics/sample_webhook.md).
+
+### How to deploy
```

Comment:
Here it would help to mention that webhook feature can be enabled or disabled using the 'webhook' flag and when it would make sense to "disable" this.

Reply:
Got it!